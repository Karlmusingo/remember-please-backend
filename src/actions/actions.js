import axios from "axios";

export const fetchEvents = () => async dispatch => {
  const { data: payload } = await axios.get(`http://localhost:5000/events/`);

  return dispatch({
    type: "FETCH_EVENTS",
    payload
  });
};
export const addEvent = event => async () => {
  try {
    const day = event.date.split("/")[0];
    const month = event.date.split("/")[1];
    const year = event.date.split("/")[2] || 2019;
    const target = event.name;
    const phonenumber = event.whatsappNumber;
    const data = {
      day,
      month,
      year,
      target,
      message: event.message,
      phonenumber,
      country: "Rwanda",
      notificationTime: 7,
      type: "Birthday"
    };

    const res = await axios.post("http://localhost:5000/events/", data);

    // console.log("res body:", res.data);

    return res.data;
  } catch (error) {
    return { error: "Oops" };
  }
};
