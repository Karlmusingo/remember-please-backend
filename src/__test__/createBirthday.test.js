import React from "react";
import { shallow } from "enzyme";
import mockAxios from "axios";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import promiseMiddleware from "redux-promise-middleware";
import { fetchEvents, addEvent } from "../actions/actions";
import BirthdayList from "../components/birthdayList";
import { jsxText } from "@babel/types";

const mockStore = configureMockStore([thunk]);

describe("Create birthday component", () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      events: {}
    });
  });

  test("Should match the snapshot", async () => {
    mockAxios.post.mockImplementationOnce(() =>
      Promise.resolve({
        data: { id: 1, name: "karl" }
      })
    );

    const events = [
      {
        date: "12/04/2019",
        name: "karl",
        message: "message",
        whatsappNumber: "+23453456456"
      }
    ];

    await store.dispatch(fetchEvents(events));
    const actions = store.getActions();

    expect.assertions(3);
    expect(actions[0].type).toEqual("FETCH_EVENTS");
    expect(actions[0].payload).toEqual("mocked");

    expect(1).toBe(1);
  });
  test("Should match the snapshot", async () => {
    mockAxios.post.mockImplementationOnce(() =>
      Promise.resolve({
        data: { id: 1, name: "karl" }
      })
    );

    const events = {
      date: "12/04/2019",
      name: "karl",
      message: "message",
      whatsappNumber: "+23453456456"
    };
    const result = await store.dispatch(addEvent(events));

    // expect.assertions(3);
    expect(result.id).toEqual(1);
    expect(result.name).toEqual("karl");

    expect(1).toBe(1);
  });

  test("Should match the snapshot", async () => {
    mockAxios.post.mockImplementationOnce(() =>
      Promise.reject({
        error: "Oops ):"
      })
    );

    const events = {
      name: "karl",
      message: "message",
      whatsappNumber: "+23453456456"
    };
    const result = await store.dispatch(addEvent(events));

    // expect.assertions(3);
    expect(result.error).toEqual("Oops");

    expect(1).toBe(1);
  });
});

describe("BirthdayList snapshot", function() {
  let store;

  beforeEach(() => {
    store = mockStore({
      events: {}
    });
  });
  test("Should match the snapshot", () => {
    const wrapper = shallow(<BirthdayList store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
describe("BirthdayList snapshot with props", function() {
  let store;

  const events = {
    todays: {
      events: [
        {
          Messages: [
            {
              draft: true
            },
            {
              draft: false
            }
          ]
        }
      ],
      eventsCount: 0
    },
    thisWeek: {
      events: [],
      eventsCount: 0
    },
    other: {
      events: [],
      eventsCount: 0
    }
  };

  beforeEach(() => {
    store = mockStore({
      events,
      onFetchEvents: jest.fn()
    });
  });
  test("Should match the snapshot", () => {
    const wrapper = shallow(<BirthdayList store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
