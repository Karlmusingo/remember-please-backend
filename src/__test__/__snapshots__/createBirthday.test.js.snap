// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirthdayList snapshot Should match the snapshot 1`] = `
<ContextProvider
  value={null}
>
  <BirthdayList
    events={Object {}}
    onFetchEvents={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;

exports[`BirthdayList snapshot with props Should match the snapshot 1`] = `
<ContextProvider
  value={null}
>
  <BirthdayList
    events={
      Object {
        "other": Object {
          "events": Array [],
          "eventsCount": 0,
        },
        "thisWeek": Object {
          "events": Array [],
          "eventsCount": 0,
        },
        "todays": Object {
          "events": Array [
            Object {
              "Messages": Array [
                Object {
                  "draft": true,
                },
                Object {
                  "draft": false,
                },
              ],
            },
          ],
          "eventsCount": 0,
        },
      }
    }
    onFetchEvents={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
